superset/assets/src/dashboard/components/ChartModal.jsx

```diff
@@ -0,0 +1,127 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { Modal } from 'react-bootstrap';
+import PropTypes from 'prop-types';
+import ChartContainer from '../../chart/ChartContainer';
+import { chartPropShape } from '../util/propShapes';
+const propTypes = {
+  showModal: PropTypes.bool,
+  animation: PropTypes.bool,
+  modalTitle: PropTypes.node,
+  modalBody: PropTypes.node,  // not required because it can be generated by beforeOpen
+  close: PropTypes.func.isRequired,
+  id: PropTypes.number,
+  width: PropTypes.number,
+  height: PropTypes.number,
+
+  chart: PropTypes.shape(chartPropShape).isRequired,
+  timeout: PropTypes.number,
+  addFilter: PropTypes.func.isRequired,
+  datasource: PropTypes.object.isRequired,
+  chartStatus: PropTypes.string,
+
+};
+const BLANK = {};
+const defaultProps = {
+  width: 550,
+  height: 350,
+  animation: true,
+  modalTitle: 'Details',
+  showModal: false,
+  addFilter: () => BLANK,
+  chart: undefined,
+  datasource: undefined,
+  timeout: 60,
+  chartStatus: 'loading',
+};
+
+export default class ChartModal extends React.PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      showModal: false,
+    };
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    if (nextState.showModal != this.state.showModal) {
+      return true;
+    }
+    if (nextProps.chartStatus != this.props.chartStatus) {
```

Comment:
should not update component for each chartStatus change

Reply:
we are changing status only at two time--
open popup-- lodaing
data recieved -- renedered

it is required to update chart from loading to rendered once data recieved